packages/ember-intl/addon/services/intl.ts

```diff
@@ -0,0 +1,397 @@
+
+    if (hasLocaleChanged(proposedLocale, this._locale)) {
+      this._locale = proposedLocale;
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore: Argument of type 'Timer | undefined' is not assignable to parameter of type 'EmberRunTimer | undefined'
+      // eslint-disable-next-line ember/no-runloop
+      cancel(this._timer);
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore: Argument of type 'Timer | undefined' is not assignable to parameter of type 'EmberRunTimer | undefined'
+      // eslint-disable-next-line ember/no-runloop
+      this._timer = next(() => {
+        this._eventEmitter.emit('localeChanged');
+
+        this.updateDocumentLanguage();
+      });
+    }
+
+    this.updateIntl(locale);
+  }
+
+  t(
+    key: string,
+    options?: FormatMessageParameters[1] & {
+      htmlSafe?: boolean;
+      locale?: string;
+    },
+  ): string {
+    const locales = options?.locale ? [options.locale] : this._locale!;
+
+    let translation: string | undefined;
+
+    for (let i = 0; i < locales!.length; i++) {
+      translation = this.getTranslation(key, locales[i]!);
+
+      if (translation !== undefined) {
+        break;
+      }
+    }
+
+    if (translation === undefined) {
+      // @ts-expect-error: Property 'resolveRegistration' does not exist on type 'Owner'
+      const missingMessage = getOwner(this).resolveRegistration(
+        'util:intl/missing-message',
+      );
+
+      return missingMessage.call(this, key, this._locale!, options) as string;
+    }
+
+    // Bypass @formatjs/intl
+    if (translation === '') {
+      return '';
+    }
+
+    return this.formatMessage(
+      {
+        defaultMessage: translation,
+        id: key,
+      },
+      options,
+    );
+  }
+
+  private createIntl(
+    locale: string | string[],
+    messages: Record<string, unknown> = {},
+  ) {
+    const resolvedLocale = convertToString(locale);
+    const formats = this._formats;
+
+    return createIntl(
+      {
+        defaultFormats: formats,
+        defaultLocale: resolvedLocale,
+        formats,
+        locale: resolvedLocale,
+        // @ts-expect-error: Type 'Record<string, unknown>' is not assignable
+        messages,
+        onError: this.onFormatjsError,
+      },
+      this._cache,
+    );
+  }
+
+  private getIntl(locale: string | string[]) {
+    const resolvedLocale = convertToString(locale);
+
+    return this._intls[resolvedLocale];
+  }
+
+  private getIntlShape(locale?: string) {
+    if (locale) {
+      return this.createIntl(locale)!;
+    }
+
+    return this.getIntl(this._locale!)!;
+  }
+
+  private getTranslation(key: string, locale: string): string | undefined {
+    const messages = this.getIntl(locale)?.messages;
+
+    if (!messages) {
+      return;
+    }
+
+    const translation = (messages as unknown as Record<string, string>)[key] as
+      | string
+      | undefined;
+
+    return translation;
+  }
```

Comment:
**breaking**

The code, now private, used to be a part of [`lookup()`](https://github.com/ember-intl/ember-intl/blob/v7.0.0-beta.5/packages/ember-intl/addon/services/intl.js#L146-L160) (public, now removed).